contracts/BaseLaunchpeg.sol

```diff
@@ -379,8 +395,38 @@ abstract contract BaseLaunchpeg is
         );
     }
 
+    /// @notice Set the public sale start time. Can only be set after phases
+    /// have been initialized.
+    /// @dev Only callable by owner
+    /// @param _publicSaleStartTime new public sale start time
+    function setPublicSaleStartTime(uint256 _publicSaleStartTime)
+        external
+        override
+        onlyOwner
+    {
+        _setPublicSaleStartTime(_publicSaleStartTime);
+    }
+
+    /// @notice Set the public sale start time. Can only be set after phases
+    /// have been initialized.
+    /// @param _publicSaleStartTime new public sale start time
+    function _setPublicSaleStartTime(uint256 _publicSaleStartTime) internal {
```

Comment:
You can put these functions `private` instead of `internal` since they are only used in the same contract.